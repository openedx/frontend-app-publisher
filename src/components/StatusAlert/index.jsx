import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { StatusAlert as Alert, Icon } from '@edx/paragon';

const StatusAlert = (props) => {
  const {
    alertType,
    className,
    iconClassNames,
    title,
    message,
    dismissible,
    onClose,
  } = props;

  return (
    <Alert
      className={className}
      alertType={alertType}
      dismissible={dismissible}
      dialog={(
        <div className={
          classNames({
            'd-flex': iconClassNames.length > 0,
          })
}
        >
          {iconClassNames.length > 0
            && (
            <div className="icon mr-2">
              <Icon className={iconClassNames} />
            </div>
            )}
          <div className="message">
            {title
              && <span className="title">{title}</span>}
            {message}
          </div>
        </div>
      )}
      onClose={onClose}
      open
    />
  );
};

StatusAlert.propTypes = {
  alertType: PropTypes.string.isRequired,
  message: PropTypes.oneOfType([
    PropTypes.string, PropTypes.element, PropTypes.array,
  ]).isRequired,
  className: PropTypes.string,
  iconClassNames: PropTypes.arrayOf(PropTypes.string),
  title: PropTypes.string,
  dismissible: PropTypes.bool,
  onClose: PropTypes.func,
};

StatusAlert.defaultProps = {
  className: '',
  iconClassNames: [],
  title: null,
  dismissible: false,
  onClose: () => {},
};

export default StatusAlert;
