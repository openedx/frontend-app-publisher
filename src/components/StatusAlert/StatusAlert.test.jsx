import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import StatusAlert from './index';

describe('StatusAlert', () => {
  it('shows a status alert', () => {
    const component = shallow(<StatusAlert
      alertType="Test"
      message="Test Message"
    />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
