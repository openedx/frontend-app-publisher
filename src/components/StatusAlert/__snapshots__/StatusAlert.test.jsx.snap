// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusAlert shows a status alert 1`] = `
<withDeprecatedProps(StatusAlert)
  alertType="Test"
  className=""
  dialog={
    <div
      className=""
    >
      <div
        className="message"
      >
        Test Message
      </div>
    </div>
  }
  dismissible={false}
  onClose={[Function]}
  open={true}
/>
`;
